packages/terra-dev-site/src/layouts/secondary-navigation-layout/SecondaryNavigationLayout.jsx

```diff
@@ -0,0 +1,353 @@
+    danglingPortalKeys.forEach((pageKey) => {
+      delete pageContainerPortalsRef.current[pageKey];
+    });
+  }, [navigationItems, pageContainerPortalsRef]);
+
+  const lastNavigationPanelOpenState = React.useRef(sideNavOverlayIsVisible);
+  React.useEffect(() => {
+    if (sideNavOverlayIsVisible && !lastNavigationPanelOpenState.current) {
+      deferExecution(() => { sideNavPanelRef.current.focus(); });
+    } else if (!sideNavOverlayIsVisible && lastNavigationPanelOpenState.current) {
+      deferExecution(() => {
+        const mainElement = document.querySelector('main');
+        if (mainElement) {
+          mainElement.focus();
+        }
+      });
+    }
+
+    lastNavigationPanelOpenState.current = sideNavOverlayIsVisible;
+  }, [sideNavOverlayIsVisible]);
+
+  React.useEffect(() => {
+    if (!sideNavOverlayIsVisible) {
+      return undefined;
+    }
+
+    function handleKeydown(e) {
+      if (e.keyCode === KEY_ESCAPE) {
+        if (e.target === pageContainerRef.current || pageContainerRef.current.contains(e.target)) {
+          setSideNavOverlayIsVisible(false);
+        }
+      }
+    }
+
+    document.addEventListener('keydown', handleKeydown);
+
+    return () => {
+      document.removeEventListener('keydown', handleKeydown);
+    };
+  }, [sideNavOverlayIsVisible, setSideNavOverlayIsVisible]);
+
+  function activatePage(pageKey) {
+    setSideNavOverlayIsVisible(false);
+
+    if (pageKey === activeNavigationKey) {
+      return;
+    }
+
+    onSelectNavigationItem(pageKey);
+  }
+
+  function renderNavigationItems() {
+    return navigationItems.map((item) => {
+      let portalElement = pageContainerPortalsRef.current[item.props.navigationKey]?.element;
+      if (!portalElement) {
+        portalElement = document.createElement('div');
+        portalElement.style.position = 'relative';
+        portalElement.style.height = '100%';
+        portalElement.style.width = '100%';
+        portalElement.id = `side-nav-${item.props.navigationKey}`;
+        pageContainerPortalsRef.current[item.props.navigationKey] = {
+          element: portalElement,
+        };
+      }
+
+      /**
+       * The cloned element is wrapped in a keyed fragment to ensure the render order of
+       * the mapped items.
+       */
+      return (
+        <React.Fragment key={item.props.navigationKey}>
+          {React.cloneElement(item, {
+            isActive: item.props.navigationKey === activeNavigationKey, portalElement,
+          })}
+        </React.Fragment>
+      );
+    });
+  }
+
+  function buildSideNavItems(childComponents) {
+    return React.Children.map(childComponents, (child) => {
+      if (child.type === NavigationItem) {
+        return { key: child.props.navigationKey, label: child.props.label };
+      }
+
+      if (child.type === SecondaryNavigationGroup) {
+        return { key: child.props.label, label: child.props.label, childItems: buildSideNavItems(child.props.children) };
```

Comment:
Should the key be `child.props.navigationKey`?

Reply:
Make sense to me, funnily enough they end up being the same key anyway: https://github.com/cerner/terra-application/blob/terra-dev-site-v7/packages/terra-dev-site/src/layouts/_DevSiteSecondaryNavigationLayout.jsx#L49-L50

Reply:
Actually i had to revert this, the secondary navigation group type does not have an active navigation key and you can't access the 'key'prop so back to label.